test/test_adaq4003.py

```diff
@@ -0,0 +1,73 @@
+import adi
+import pytest
+
+hardware = "adaq4003"
+classname = "adi.adaq4003"
+
+#########################################
+@pytest.mark.iio_hardware(hardware)
+@pytest.mark.parametrize("classname", [(classname)])
+@pytest.mark.parametrize(
+    "attr, val",
+    [
+        (
+            "sampling_frequency",
+            [10000, 50000, 100000, 200000, 500000, 1000000, 2000000],
+        ),
+    ],
+)
+def test_adaq4003_attr(test_attribute_multiple_values, iio_uri, classname, attr, val):
+
+    # Setting the sampling frequency is only possible through FPGA implemented
+    # hardware (SPI-Engine offload) so the attribute might not exist if the
+    # the platform is not set with proper FPGA bitstream or doesn't have
+    # a integrated FPGA.
+    # if not hasattr(sdr, attr):
+    #     raise AttributeError(attr + " not defined in " + classname)
+
+    try:
+        test_attribute_multiple_values(iio_uri, classname, attr, val, 1)
+    except KeyError:
+        pytest.skip("sampling_frequency attribute not available on test platform.")
+
+
+#########################################
+@pytest.mark.iio_hardware(hardware)
+@pytest.mark.parametrize("classname", [(classname)])
+@pytest.mark.parametrize(
+    "attr, avail_attr, tol, repeats, sub_channel",
+    [("scale", "scale_available", 0, 1, "voltage0-voltage1",),],
```

Comment:
As noted above, there is no scale_available. Also this is a single-channel device with a single differential input, voltage0. The voltage0-voltage-1 nomenclature is typical of multichannel devices that can be arbitrarily configured.
Note that I am looking at the ADAQ4003 on Cora, but I see zynq-zed-adv7511-adxxxx noted below. The ADAQ4003 attributes, channel, etc.